src/Analysis/Ast/Impl/Specializations/BuiltinsSpecializations.cs

```diff
@@ -91,6 +94,79 @@ public static IMember List(IPythonInterpreter interpreter, IPythonFunctionOverlo
             return args.Count > 0 && args[0] is PythonCollection c ? c.Contents.FirstOrDefault() : null;
         }
 
+        public static IMember Super(IPythonModule declaringModule, IPythonFunctionOverload overload, IArgumentSet argSet, IndexSpan indexSpan) {
+            var args = argSet.Values<IMember>();
+
+            if (args.Count() == 0) {
+                //Zero argument form only works inside a class definition
+                foreach (var s in argSet.Eval.CurrentScope.EnumerateTowardsGlobal) {
+                    if (s.Node is ClassDefinition) {
+                        var classType = s.Variables["__class__"].GetPythonType<IPythonClassType>();
+
+                        if (classType?.Mro != null) {
+                            return CreateSuper(argSet, classType.Mro, declaringModule, indexSpan);
+                        }
+                    }
+                }
+                return null;
+            }
+
+            // If multiple arguments first argument is required
+            var firstCls = args.FirstOrDefault().GetPythonType<IPythonClassType>();
+            if (firstCls == null) {
+                return null;
+            }
+
+            // second argument optional
+            bool isUnbound = args.Count() == 1;
```

Comment:
do we know whether IReadOnlyList is treated special in Linq? I just checked (https://github.com/dotnet/corefx/blob/master/src/System.Linq/src/System/Linq/Count.cs) and it is not. meaning this use Enumerator unnecessarily since IReadOnlyList provide indexer and Count property?

Reply:
good point. I'll avoid Count() and use .Count instead

Reply:
updated